### PR TITLE
docs: Translate Conditional Rendering

### PR DESCRIPTION
**Pull request by brian3814**
This is my first time participating in a translation project, please let me know if anything can be improved.

Thanks

### CODE REVIEW

File: src/content/learn/conditional-rendering.md [hunk 1]
```diff
@@ -177,13 +177,13 @@ if (isPacked) {
...
+雖然這個重複不會造成太大影響，但它可能會讓你的程式碼維護起來較不容易。假如你想要修改 `className` 的值呢？你會需要修改兩處程式碼！在這種情況下，你可以有條件地包含一些 JSX，讓你的程式碼更符合 [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) 的原則。
 
-### Conditional (ternary) operator (`? :`) {/*conditional-ternary-operator--*/}
+### 條件 (三元) 運算子 (`? :`) {/*conditional-ternary-operator--*/}
 
-JavaScript has a compact syntax for writing a conditional expression -- the [conditional operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) or "ternary operator".
+JavaScript 有一個簡潔的語法可以用來撰寫條件表達式--[條件運算子](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) 或「三元運算子」。
 
-Instead of this:
+由此種寫法：
```
Comment (neighborhood999) [inline src/content/learn/conditional-rendering.md:186]:
```suggestion
不使用這種寫法：
```

File: src/content/learn/conditional-rendering.md [hunk 2]
```diff
@@ -202,17 +202,17 @@ return (
...
-You can read it as *"if `isPacked` is true, then (`?`) render `name + ' ✔'`, otherwise (`:`) render `name`"*.
+你可以把這段程式碼解讀為：「如果 `isPacked` 的值為 `true`，則 (`?`) render `name + ' ✔'`；否則 (`:`)，render `name` 」。
 
 <DeepDive>
 
-#### Are these two examples fully equivalent? {/*are-these-two-examples-fully-equivalent*/}
+#### 上述兩個範例是完全等價的嗎？ {/*are-these-two-examples-fully-equivalent*/}
 
-If you're coming from an object-oriented programming background, you might assume that the two examples above are subtly different because one of them may create two different "instances" of `<li>`. But JSX elements aren't "instances" because they don't hold any internal state and aren't real DOM nodes. They're lightweight descriptions, like blueprints. So these two examples, in fact, *are* completely equivalent. [Preserving and Resetting State](/learn/preserving-and-resetting-state) goes into detail about how this works.
+如果你有物件導向開發的相關背景，你或許會因為其中一個範例有可能會建立兩個不同的 `<li>` instance，從而認為上面的兩個範例略有不同。但 JSX 元素其實並不是 instance，因為它們沒有保存任何內部狀態，也不是真正的 DOM 節點。它們是像藍圖一樣的輕量化描述。所以上述兩個例子事實上 *是* 完全等價的。 [保存與重設狀態](/learn/preserving-and-resetting-state) 中詳細介紹了它的運作原理。
```
Comment (neighborhood999) [inline src/content/learn/conditional-rendering.md:211]: I think `保留` is better than `保存`, any ideas? 😃

```suggestion
如果你有物件導向開發的相關背景，你或許會因為其中一個範例有可能會建立兩個不同的 `<li>` instance，從而認為上面的兩個範例略有不同。但 JSX 元素其實並不是 instance，因為它們沒有保存任何內部狀態，也不是真正的 DOM 節點。它們是像藍圖一樣的輕量化描述。所以上述兩個例子事實上 是* 全等價的。[保留與重設狀態](/learn/preserving-and-resetting-state) 詳細介紹了它的運作原理。
```

File: src/content/learn/conditional-rendering.md [hunk 3]
```diff
@@ -310,46 +310,46 @@ export default function PackingList() {
 
 </Sandpack>
 
-A [JavaScript && expression](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND) returns the value of its right side (in our case, the checkmark) if the left side (our condition) is `true`. But if the condition is `false`, the whole expression becomes `false`. React considers `false` as a "hole" in the JSX tree, just like `null` or `undefined`, and doesn't render anything in its place.
+[JavaScript && 運算子](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND) 會在左側的條件為 `true` 時返回它右側的值（在範例中是勾號）。但如果條件為 `false`，整個表達式的結果就會變成 `false`。跟 `null` 或 `undefined` 一樣，React 會將 `false` 視為像是 JSX tree 中的一個 「洞」，因此不會 render 任何東西。
```
Comment (neighborhood999) [inline src/content/learn/conditional-rendering.md:313]:
```suggestion
[JavaScript && 運算子](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND) 會在左側的條件為 `true` 時回傳它右側的值（在範例中是勾號）。但如果條件為 `false`，整個表達式的結果就會變成 `false`。跟 `null` 或 `undefined` 一樣，React 會將 `false` 視為像是 JSX tree 中的一個 「洞」，因此不會 render 任何東西。
```

File: src/content/learn/conditional-rendering.md [hunk 4]
```diff
@@ -310,46 +310,46 @@ export default function PackingList() {
...
+接著在 `isPacked` 的值為 `true` 時，使用 `if` 陳述式重新將 `itemContent` 的值指定為一個 JSX 表達式。 
 
 ```js
 if (isPacked) {
   itemContent = name + " ✔";
 }
 ```
 
-[Curly braces open the "window into JavaScript".](/learn/javascript-in-jsx-with-curly-braces#using-curly-braces-a-window-into-the-javascript-world) Embed the variable with curly braces in the returned JSX tree, nesting the previously calculated expression inside of JSX:
+[大括弧打開了引入 Javascript 的大門。](/learn/javascript-in-jsx-with-curly-braces#using-curly-braces-a-window-into-the-javascript-world) 藉由大括弧可以將變數嵌入回傳的 JSX tree 中，進而將先前計算的表達式巢狀地嵌在 JSX 中：
```
Comment (neighborhood999) [inline src/content/learn/conditional-rendering.md:344]:
```suggestion
[大括弧打開了引入 JavaScript 的大門。](/learn/javascript-in-jsx-with-curly-braces#using-curly-braces-a-window-into-the-javascript-world) 由大括弧可以將變數嵌入回傳的 JSX tree 中，進而將先前計算的表達式巢狀地嵌在 JSX 中：
```

File: src/content/learn/conditional-rendering.md [hunk 5]
```diff
@@ -624,15 +624,15 @@ export default function PackingList() {
 
 </Sandpack>
 
-Note that you must write `importance > 0 && ...` rather than `importance && ...` so that if the `importance` is `0`, `0` isn't rendered as the result!
+要注意的是你必須寫成 `importance > 0 && ...` 而不是 `importance && ...` ，這樣如果 `importance` 的值為 `0` 時，才不會把 `0` 做為結果 render 出來。
 
-In this solution, two separate conditions are used to insert a space between the name and the importance label. Alternatively, you could use a fragment with a leading space: `importance > 0 && <> <i>...</i></>` or add a space immediately inside the `<i>`:  `importance > 0 && <i> ...</i>`.
+解答中為了在 name 和 importance 標籤之間插入一個空格，使用了兩段不同的條件。你也可以使用帶有空格前綴的 fragment： `importance > 0 && <> <i>...</i></>`，或是直接在 `<i>` 內部添加一個空格： `importance > 0 && <i> ...</i>`。
```
Comment (neighborhood999) [inline src/content/learn/conditional-rendering.md:629]:
```suggestion
解答中為了在 name 和 importance 標籤之間插入一個空格，使用了兩段不同的條件。你也可以使用帶有空格前綴的 fragment： `importance > 0 && <> <i>...</i></>`，或是直接在 `<i>` 內部增加一個空格： `importance > 0 && <i> ...</i>`。
```

File: src/content/learn/conditional-rendering.md [hunk 6]
```diff
@@ -712,9 +712,9 @@ export default function DrinkList() {
 
 </Sandpack>
 
-Here the information about each drink is grouped together instead of being spread across multiple conditions. This makes it easier to add more drinks in the future.
+在這個做法中，每個飲料的相關資訊被集中在了一起，而非分散在多個條件語句中。這麼做可以讓未來在添加更多的飲料時較為方便。
```
Comment (neighborhood999) [inline src/content/learn/conditional-rendering.md:715]:
```suggestion
在這個做法中，每個飲料的相關資訊被集中在了一起，而非分散在多個條件語句中。這麼做可以讓未來在增加更多的飲料時較為方便。
```

File: src/content/learn/conditional-rendering.md (hunk 2 above)
Comment (brian3814) [reply src/content/learn/conditional-rendering.md:211]: Sounds good to me! Will change it to `保留`.

**Review by neighborhood999 — APPROVED** [review]

**Pull request merged**